src/js/control-bar/play-toggle.js

```diff
@@ -58,6 +58,12 @@ class PlayToggle extends Button {
    */
   handleClick(event) {
     if (this.player_.paused()) {
+      // Safari will not restart playback on a play click when playing back natively, so we seek to 0, then play.
+      const endedNativePlayback = this.hasClass('vjs-ended') && !this.player_.lastSource_.tech.startsWith('blob');
```

Comment:
I don't think `lastSource_` will always be present. With the YouTube tech for example.

Reply:
Moved everything to `player.js` per @gkatsev's suggestion and used the `browser` constants to check for native playback rather than the `lastSource_`